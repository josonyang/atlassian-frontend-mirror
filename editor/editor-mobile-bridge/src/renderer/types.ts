import {
	type AnnotationMarkStates,
	type AnnotationId,
	type AnnotationTypes,
} from '@atlaskit/adf-schema';

export type AnnotationPayload = {
	annotationId: AnnotationId;
	annotationType: AnnotationTypes;
};

export type AnnotationStatePayload = {
	annotationId: AnnotationId;
	annotationType: AnnotationTypes;
	annotationState: AnnotationMarkStates;
};
