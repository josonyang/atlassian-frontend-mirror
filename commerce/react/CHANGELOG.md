# @af/commerce-react

## 0.1.3

### Patch Changes

- [`61cb5313358`](https://bitbucket.org/atlassian/atlassian-frontend/commits/61cb5313358) - Removing unused dependencies and dev dependencies

## 0.1.2

### Patch Changes

- [`9d00501a414`](https://bitbucket.org/atlassian/atlassian-frontend/commits/9d00501a414) - Ensure legacy types are published for TS 4.5-4.8

## 0.1.1

### Patch Changes

- [`41fae2c6f68`](https://bitbucket.org/atlassian/atlassian-frontend/commits/41fae2c6f68) - Upgrade Typescript from `4.5.5` to `4.9.5`

## 0.1.0

### Minor Changes

- [`56507598609`](https://bitbucket.org/atlassian/atlassian-frontend/commits/56507598609) - Skip minor dependency bump

## 0.0.5

### Patch Changes

- [`5806023831a`](https://bitbucket.org/atlassian/atlassian-frontend/commits/5806023831a) - Internal change to enforce token usage for spacing properties. There is no expected visual or behaviour change.

## 0.0.4

### Patch Changes

- [`8cc2f888c83`](https://bitbucket.org/atlassian/atlassian-frontend/commits/8cc2f888c83) - Upgrade Typescript from `4.3.5` to `4.5.5`

## 0.0.3

### Patch Changes

- [`8d4228767b0`](https://bitbucket.org/atlassian/atlassian-frontend/commits/8d4228767b0) - Upgrade Typescript from `4.2.4` to `4.3.5`.

## 0.0.2

### Patch Changes

- [`cb2392f6d33`](https://bitbucket.org/atlassian/atlassian-frontend/commits/cb2392f6d33) - Upgrade to TypeScript 4.2.4

## 0.0.1

### Patch Changes

- [`e5c858ab40c`](https://bitbucket.org/atlassian/atlassian-frontend/commits/e5c858ab40c) - Updating stripe and managing state update

## 0.0.4

### Patch Changes

- [`229b32842b5`](https://bitbucket.org/atlassian/atlassian-frontend/commits/229b32842b5) - Fix .npmignore and tsconfig.json for **tests**

## 0.0.3

### Patch Changes

- [`d3265f19be`](https://bitbucket.org/atlassian/atlassian-frontend/commits/d3265f19be) - Transpile packages using babel rather than tsc

## 0.0.2

### Patch Changes

- [`5f58283e1f`](https://bitbucket.org/atlassian/atlassian-frontend/commits/5f58283e1f) - Export types using Typescript's new "export type" syntax to satisfy Typescript's --isolatedModules compiler option.
  This requires version 3.8 of Typescript, read more about how we handle Typescript versions here: https://atlaskit.atlassian.com/get-started
  Also add `typescript` to `devDependencies` to denote version that the package was built with.

## 0.0.1

### Patch Changes

- [`b443b5a60f`](https://bitbucket.org/atlassian/atlassian-frontend/commits/b443b5a60f) - Renamed template package
