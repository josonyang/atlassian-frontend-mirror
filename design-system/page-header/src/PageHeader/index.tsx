// TODO: Remove this eslint-disable when prop names have been renamed.
// This rule is here as prop name changes are a major as they are used
// by our consumers. This can be done in the next lite-mode conversion.
/* eslint-disable @repo/internal/react/boolean-prop-naming-convention */
import React, { type ReactElement, type ReactNode } from 'react';

import {
	ActionsWrapper,
	BottomBarWrapper,
	OuterWrapper,
	StyledTitleWrapper,
	TitleContainer,
	TitleWrapper,
} from './styled';

type PageHeaderProps = {
	/**
	 * Contents of the action bar to be rendered next to the page title.
	 */
	actions?: ReactElement;
	/**
	 * Contents of the action bar to be rendered next to the page title. Typically a button group.
	 */
	bottomBar?: ReactElement;
	/**
	 * Page breadcrumbs to be rendered above the title.
	 */
	breadcrumbs?: ReactElement;
	/**
	 * Contents of the bottom bar to be rendered below the page title. Typically contains a search bar and/or filters.
	 */
	children?: ReactNode;
	/**
	 * Content of the page title. The text wraps by default.
	 */
	disableTitleStyles?: boolean;
	/**
	 * Returns the inner ref to the DOM element of the title. This is exposed so the focus can be set.
	 */
	innerRef?: (element: HTMLElement) => void;
	/**
	 * Prevent the title from wrapping across lines. Avoid using this wherever possible, as truncation can make page headings inaccessible.
	 */
	truncateTitle?: boolean;
	/**
	 * Used as the ID of the inner h1 tag. This is exposed so the header text can be used as label of other elements by aria-labelledby.
	 */
	id?: string;
	/**
	 * A testId prop is provided for specified elements, which is a unique string that appears as a data attribute data-testid in the rendered code, serving as a hook for automated tests.
	 */
	testId?: string;
};

/**
 * __Page header__
 *
 * A page header defines the top of a page. It contains a title and can be optionally combined with
 * breadcrumbs buttons, search, and filters.
 *
 * - [Examples](https://atlassian.design/components/page-header/examples)
 * - [Code](https://atlassian.design/components/page-header/code)
 * - [Usage](https://atlassian.design/components/page-header/usage)
 */
const PageHeader = ({
	innerRef,
	breadcrumbs,
	actions,
	bottomBar,
	children,
	id,
	disableTitleStyles = false,
	truncateTitle = false,
	testId,
}: PageHeaderProps) => {
	return (
		<OuterWrapper>
			{breadcrumbs}
			<TitleWrapper truncateTitle={truncateTitle}>
				<TitleContainer truncateTitle={truncateTitle}>
					{disableTitleStyles ? (
						children
					) : (
						<StyledTitleWrapper
							ref={innerRef as React.Ref<HTMLHeadingElement>}
							truncateTitle={truncateTitle}
							id={id}
							testId={testId}
						>
							{children}
						</StyledTitleWrapper>
					)}
				</TitleContainer>
				{actions && <ActionsWrapper>{actions}</ActionsWrapper>}
			</TitleWrapper>
			{bottomBar && <BottomBarWrapper> {bottomBar} </BottomBarWrapper>}
		</OuterWrapper>
	);
};

export default PageHeader;
