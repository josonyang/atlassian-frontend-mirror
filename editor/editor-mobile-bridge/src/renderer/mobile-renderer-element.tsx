/* eslint-disable no-console */
import React, { useCallback, useEffect, useRef } from 'react';
import type { AnnotationProviders } from '@atlaskit/editor-common/types';
import type {
  ExtensionHandlers,
  ExtensionProvider,
} from '@atlaskit/editor-common/extensions';
import { WithCreateAnalyticsEvent } from '@atlaskit/editor-common/ui';
import type { CreateUIAnalyticsEvent } from '@atlaskit/analytics-next';
import type { MentionProvider } from '@atlaskit/mention/types';
import type { MediaProvider as MediaProviderType } from '@atlaskit/editor-common/provider-factory';
import type { RendererProps } from '@atlaskit/renderer';
import { ReactRenderer } from '@atlaskit/renderer';
import FabricAnalyticsListeners from '@atlaskit/analytics-listeners';
import { toNativeBridge } from './web-to-native/implementation';
import type { CardClient } from '@atlaskit/link-provider';
import { SmartCardProvider } from '@atlaskit/link-provider';
import type { EmojiResource } from '@atlaskit/emoji/resource';
import {
  getEnableLightDarkTheming,
  getAllowCaptions,
  getEnableTokenThemes,
} from '../query-param-reader';
import { rendererAnalyticsClient } from './renderer-analytics-client';
import { useRendererContent } from './hooks/use-set-renderer-content';
import { useCreateProviderFactory } from './hooks/use-create-provider-factory';
import { useRendererContext } from './hooks/use-renderer-context';
import { useHeadingLinks } from './hooks/use-heading-links';
import { useAnnotation } from './annotations/use-annotation';
import { useRendererReady } from './hooks/use-renderer-ready';
import { useRendererDestroyed } from './hooks/use-renderer-destroyed';
import { eventHandlers } from './event-handlers';
import { isApple } from '../utils/is-apple';
import { useRendererReflowDetected } from './hooks/use-renderer-reflow-detected';

import { withIntlProvider } from '../i18n/with-intl-provider';
import type { IntlShape } from 'react-intl-next';
import { injectIntl } from 'react-intl-next';
import { geti18NMessages } from './renderer-localisation-provider';
import { withSystemTheme } from '../WithSystemTheme';
import type RendererBridgeImplementation from './native-to-web/implementation';
import type { DocNode } from '@atlaskit/adf-schema';
import { setGlobalTheme } from '@atlaskit/tokens';

export interface MobileRendererProps extends RendererProps {
  cardClient: CardClient;
  disableMediaLinking?: boolean;
  document: DocNode;
  emojiProvider: Promise<EmojiResource>;
  intl: IntlShape;
  mediaProvider: Promise<MediaProviderType>;
  mentionProvider: Promise<MentionProvider>;
  extensionProvider?: Promise<ExtensionProvider>;
  rendererBridge: RendererBridgeImplementation;
}

type WithSmartCardClientProps = {
  cardClient: CardClient;
};

type BasicRendererProps = {
  allowAnnotations: boolean;
  allowHeadingAnchorLinks: boolean;
  containerAri: string;
  disableMediaLinking: boolean;
  disableActions: boolean;
  document: DocNode;
  emojiProvider: Promise<EmojiResource>;
  extensionHandlers: ExtensionHandlers;
  intl: IntlShape;
  mediaProvider: Promise<MediaProviderType>;
  mentionProvider: Promise<MentionProvider>;
  extensionProvider?: Promise<ExtensionProvider>;
  objectAri: string;
  rendererBridge: RendererBridgeImplementation;
  allowCustomPanels: boolean;
};

interface WithCreateAnalyticsEventProps extends BasicRendererProps {
  createAnalyticsEvent: CreateUIAnalyticsEvent;
  annotationProvider: AnnotationProviders | null;
}

const handleRendererContentLoadedBridge = () => {
  if (
    !isApple(window) && // don't fire on iOS
    window.requestAnimationFrame
  ) {
    window.requestAnimationFrame(() =>
      toNativeBridge.call('renderBridge', 'onContentRendered'),
    );
  }
};

const BasicRenderer: React.FC<WithCreateAnalyticsEventProps> = ({
  allowAnnotations,
  allowHeadingAnchorLinks,
  createAnalyticsEvent,
  disableActions,
  disableMediaLinking,
  document: initialDocument,
  emojiProvider,
  mediaProvider,
  mentionProvider,
  extensionProvider,
  rendererBridge,
  allowCustomPanels,
}: WithCreateAnalyticsEventProps) => {
  const document = useRendererContent(initialDocument);
  let providers = {
    mentionProvider,
    emojiProvider,
    mediaProvider,
    extensionProvider,
  };
  const providerFactory = useCreateProviderFactory(providers, rendererBridge);
  const rendererContext = useRendererContext(rendererBridge);
  const headingAnchorLinksConfig = useHeadingLinks(allowHeadingAnchorLinks);
  const annotationProvider = useAnnotation(allowAnnotations);
  const innerRef = useRef<HTMLDivElement>(null);

  useRendererReady(innerRef);
  useRendererDestroyed();
  useRendererReflowDetected(rendererBridge);

  useEffect(() => {
    const setTheme = async () => await setGlobalTheme({ colorMode: 'auto' });
    if (getEnableTokenThemes()) {
      setTheme();
    }
  }, []);

  return (
    <ReactRenderer
      innerRef={innerRef}
      adfStage="stage0"
      document={document}
      annotationProvider={annotationProvider}
      allowAnnotations={allowAnnotations}
      dataProviders={providerFactory}
      onError={handleRendererContentLoadedBridge}
      onComplete={handleRendererContentLoadedBridge}
      appearance="mobile"
      disableActions={disableActions}
      createAnalyticsEvent={createAnalyticsEvent}
      allowAltTextOnImages
      media={{
        allowLinking: !disableMediaLinking,
        allowCaptions: getAllowCaptions(),
      }}
      allowHeadingAnchorLinks={headingAnchorLinksConfig}
      rendererContext={rendererContext}
      eventHandlers={eventHandlers}
      useSpecBasedValidator={true}
      allowCustomPanels={allowCustomPanels}
    />
  );
};

const withSmartCard =
  <P extends BasicRendererProps>(
    Component: React.ComponentType<P>,
  ): React.FC<P & WithSmartCardClientProps> =>
  ({ cardClient: smartCardClient, ...props }: WithSmartCardClientProps) => {
    // Temporarily opting out of the default oauth2 flow for phase 1 of Smart Links
    // See https://product-fabric.atlassian.net/browse/FM-2149 for details.
    const authFlow = 'disabled';
    const renderCallback = useCallback(
      (createAnalyticsEvent) => (
        <SmartCardProvider client={smartCardClient} authFlow={authFlow}>
          <Component
            createAnalyticsEvent={createAnalyticsEvent}
            {...(props as P)}
          />
        </SmartCardProvider>
      ),
      [props, smartCardClient],
    );

    return <WithCreateAnalyticsEvent render={renderCallback} />;
  };

const withFabricAnalytics =
  <P extends MobileRendererProps>(
    Component: React.ComponentType<P>,
  ): React.FC<MobileRendererProps> =>
  (props: MobileRendererProps) => {
    return (
      <FabricAnalyticsListeners client={rendererAnalyticsClient}>
        <Component {...(props as P)} />
      </FabricAnalyticsListeners>
    );
  };

const ThemedBasicRenderer = getEnableTokenThemes()
  ? BasicRenderer
  : withSystemTheme(BasicRenderer, getEnableLightDarkTheming());

const MobileRenderer = withFabricAnalytics(withSmartCard(ThemedBasicRenderer));

export { MobileRenderer };

export default withIntlProvider(injectIntl(MobileRenderer), geti18NMessages);
