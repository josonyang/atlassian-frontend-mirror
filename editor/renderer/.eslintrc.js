module.exports = {
  rules: {
    '@atlaskit/design-system/ensure-design-token-usage/preview': [
      'error',
      { domains: ['spacing'], shouldEnforceFallbacks: false },
    ],
  },
};
