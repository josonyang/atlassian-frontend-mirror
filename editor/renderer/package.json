{
  "name": "@atlaskit/renderer",
  "version": "108.11.3",
  "description": "Renderer component",
  "publishConfig": {
    "registry": "https://registry.npmjs.org/"
  },
  "repository": "https://bitbucket.org/atlassian/atlassian-frontend",
  "author": "Atlassian Pty Ltd",
  "license": "Apache-2.0",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "module:es2019": "dist/es2019/index.js",
  "types": "dist/types/index.d.ts",
  "sideEffects": false,
  "atlaskit:src": "src/index.ts",
  "atlassian": {
    "team": "Editor",
    "deprecatedAutoEntryPoints": true,
    "inPublicMirror": true,
    "releaseModel": "continuous",
    "website": {
      "name": "Renderer"
    }
  },
  "dependencies": {
    "@atlaskit/adf-schema": "^28.1.0",
    "@atlaskit/adf-utils": "^19.0.0",
    "@atlaskit/analytics-listeners": "^8.7.0",
    "@atlaskit/analytics-namespaced-context": "^6.7.0",
    "@atlaskit/analytics-next": "^9.1.0",
    "@atlaskit/button": "^16.9.0",
    "@atlaskit/code": "^14.6.0",
    "@atlaskit/editor-common": "^74.45.0",
    "@atlaskit/editor-json-transformer": "^8.10.0",
    "@atlaskit/editor-palette": "1.5.1",
    "@atlaskit/editor-prosemirror": "1.1.0",
    "@atlaskit/editor-shared-styles": "^2.6.0",
    "@atlaskit/emoji": "^67.4.0",
    "@atlaskit/icon": "^21.12.0",
    "@atlaskit/link-datasource": "^0.33.0",
    "@atlaskit/media-card": "^76.1.0",
    "@atlaskit/media-client": "^23.1.0",
    "@atlaskit/media-common": "^8.0.0",
    "@atlaskit/media-filmstrip": "^47.0.0",
    "@atlaskit/media-ui": "^23.3.0",
    "@atlaskit/media-viewer": "^48.0.0",
    "@atlaskit/platform-feature-flags": "^0.2.0",
    "@atlaskit/smart-card": "^26.15.0",
    "@atlaskit/status": "^1.3.0",
    "@atlaskit/task-decision": "^17.7.0",
    "@atlaskit/theme": "^12.5.0",
    "@atlaskit/tokens": "^1.14.0",
    "@atlaskit/tooltip": "^17.8.0",
    "@babel/runtime": "^7.0.0",
    "@emotion/react": "^11.7.1",
    "lodash": "^4.17.21",
    "memoize-one": "^6.0.0",
    "react-intl-next": "npm:react-intl@^5.18.1",
    "react-loadable": "^5.1.0",
    "uuid": "^3.1.0"
  },
  "peerDependencies": {
    "@atlaskit/link-provider": "^1.6.2",
    "@atlaskit/media-core": "^34.1.2",
    "react": "^16.8.0",
    "react-dom": "^16.8.0"
  },
  "devDependencies": {
    "@af/visual-regression": "*",
    "@atlaskit/analytics-gas-types": "^5.1.0",
    "@atlaskit/css-reset": "^6.5.0",
    "@atlaskit/editor-test-helpers": "^18.10.0",
    "@atlaskit/link-provider": "^1.6.0",
    "@atlaskit/link-test-helpers": "^6.0.0",
    "@atlaskit/media-core": "^34.1.0",
    "@atlaskit/media-integration-test-helpers": "^3.0.0",
    "@atlaskit/media-test-helpers": "^33.0.0",
    "@atlaskit/mention": "^22.1.0",
    "@atlaskit/navigation-next": "^9.0.0",
    "@atlaskit/util-data-test": "^17.8.0",
    "@atlaskit/visual-regression": "*",
    "@atlaskit/webdriver-runner": "*",
    "@atlassian/atlassian-frontend-prettier-config-1.0.1": "npm:@atlassian/atlassian-frontend-prettier-config@1.0.1",
    "@atlassian/feature-flags-test-utils": "*",
    "@testing-library/react": "^12.1.5",
    "@types/react-loadable": "^5.4.1",
    "ajv": "^6.12.6",
    "enzyme": "^3.10.0",
    "jsdom": "^17.0.0",
    "mockdate": "^3.0.5",
    "puppeteer": "13.7.0",
    "react": "^16.8.0",
    "react-dom": "^16.8.0",
    "react-live-clock": "^5.0.0",
    "react-moment": "^1.1.2",
    "react-test-renderer": "^16.8.0",
    "rxjs": "^5.5.0",
    "sinon": "^2.2.0",
    "typescript": "~4.9.5",
    "worker-plugin": "^4.0.2"
  },
  "prettier": "@atlassian/atlassian-frontend-prettier-config-1.0.1",
  "techstack": {
    "@repo/internal": {
      "design-tokens": [
        "color"
      ],
      "deprecation": "no-deprecated-imports",
      "styling": [
        "emotion"
      ]
    }
  },
  "platform-feature-flags": {
    "platform.editor.custom-table-width": {
      "type": "boolean"
    },
    "platform.editor.disable-default-width-table-scaling-renderer": {
      "type": "boolean"
    }
  }
}
