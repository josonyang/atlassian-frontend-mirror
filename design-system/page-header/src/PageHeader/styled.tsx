// TODO: Remove this eslint-disable when prop names have been renamed.
// This rule is here as prop name changes are a major as they are used
// by our consumers. The prop name concerned here is truncateTitle.
// This can be done in the next lite-mode conversion.
/* eslint-disable @repo/internal/react/boolean-prop-naming-convention */
/** @jsx jsx */

import React, { type ReactNode } from 'react';

import { css, jsx } from '@emotion/react';

import { getBooleanFF } from '@atlaskit/platform-feature-flags';
import { Box, xcss } from '@atlaskit/primitives';
import { h700 } from '@atlaskit/theme/typography';
import { token } from '@atlaskit/tokens';

const truncateStyles = css({
	overflowX: 'hidden',
	textOverflow: 'ellipsis',
	whiteSpace: 'nowrap',
});

const outerStyles = css({
	margin: `${token('space.300', '24px')} 0 ${token('space.200', '16px')} 0`,
});

const titleStyles = css({
	color: token('color.text'),
	font: token('font.heading.large'),
	letterSpacing: 'normal',
	marginBlockStart: 0,
	outline: 'none',
});

const styledTitleStyles = css({
	lineHeight: token('font.lineHeight.500', '32px'),
	marginBlockStart: 0,
	outline: 'none',
});

const titleWrapperStyles = css({
	display: 'flex',
	alignItems: 'flex-start',
	flexWrap: 'wrap',
});

const titleWrapperTruncateStyles = css({
	flexWrap: 'nowrap',
});

const titleContainerStyles = css({
	minWidth: 0,
	maxWidth: '100%',
	flex: '1 0 auto',
	flexShrink: undefined,
	marginBlockEnd: token('space.100', '8px'),
});

const actionStyles = css({
	maxWidth: '100%',
	flex: '0 0 auto',
	marginBlockEnd: token('space.100', '8px'),
	marginInlineStart: 'auto',
	paddingInlineStart: token('space.400', '32px'),
	whiteSpace: 'nowrap',
	// eslint-disable-next-line @atlaskit/design-system/no-nested-styles
	'>': {
		textAlign: 'right',
	},
});

const titleContainerTruncateStyles = css({
	flexShrink: 1,
});

const bottomBarStyles = xcss({
	marginBlockStart: 'space.200',
});

/**
 * __Outer wrapper__.
 *
 * An outer wrapper that is the outermost component of the PageHeader component. It wraps around the PageHeader, its Actions,
 * the BottomBar and its Breadcrumbs.
 *
 */
export const OuterWrapper = ({ children }: { children: ReactNode }) => {
	return <div css={outerStyles}>{children}</div>;
};

// eslint-disable-next-line @atlaskit/design-system/consistent-css-prop-usage
const h700Styles = css(h700());

interface StyledTitleWrapperProps {
	children?: React.ReactNode;
	id?: string;
	truncateTitle?: boolean;
}

/**
 * __Styled title wrapper__.
 *
 * A styled title wrapper is a wrapper around the title that controls its the styles exclusively.
 *
 */
export const StyledTitleWrapper = React.forwardRef<HTMLHeadingElement, StyledTitleWrapperProps>(
	({ children, id, truncateTitle }, ref) => {
		return (
			<h1
				css={[
					...(getBooleanFF(
						'platform.design-system-team.page-header-tokenised-typography-styles_lj1ix',
					)
						? [titleStyles]
						: [h700Styles, styledTitleStyles]),
					truncateTitle && truncateStyles,
				]}
				ref={ref}
				tabIndex={-1}
				id={id}
			>
				{children}
			</h1>
		);
	},
);

interface TitleProps {
	truncateTitle?: boolean;
	children?: ReactNode;
}

/**
 * __Title wrapper__.
 *
 * A title wrapper is a wrapper around the title and the actions.
 *
 */
export const TitleWrapper = ({ children, truncateTitle }: TitleProps) => {
	return (
		<div css={[titleWrapperStyles, truncateTitle && titleWrapperTruncateStyles]}>{children}</div>
	);
};

/**
 * Title container.
 *
 * A title container is a container that wraps around the title and its styles (if applied).
 *
 */
export const TitleContainer = ({ children, truncateTitle }: TitleProps) => {
	return (
		<div css={[titleContainerStyles, truncateTitle && titleContainerTruncateStyles]}>
			{children}
		</div>
	);
};

/**
 * __Actions wrapper__.
 *
 * An actions wrapper is a wrapper for the actions, which appear on the top right of the PageHeader component.
 *
 */
export const ActionsWrapper = ({ children }: { children: ReactNode }) => {
	return <div css={actionStyles}>{children}</div>;
};

/**
 * __Bottom bar wrapper__.
 *
 * A bottom bar wrapper is a wrapper for the bottom bar, which appears at the bottom of the PageHeader component.
 *
 */
export const BottomBarWrapper = ({ children }: { children: ReactNode }) => {
	return <Box xcss={bottomBarStyles}>{children}</Box>;
};
