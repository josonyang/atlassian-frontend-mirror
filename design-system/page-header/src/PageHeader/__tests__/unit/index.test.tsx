import React from 'react';

import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import PageHeader from '../../../index';

describe('@atlaskit/page-header', () => {
	it('should render correctly', () => {
		const BreadCrumbs = () => <div>Breadcrumb</div>;
		const Actions = () => <div>Action</div>;
		const Bar = () => <div>Bar</div>;

		const wrapper = (
			<PageHeader breadcrumbs={<BreadCrumbs />} actions={<Actions />} bottomBar={<Bar />}>
				Test
			</PageHeader>
		);
		const Component = renderer.create(wrapper).toJSON();
		expect(Component).toMatchSnapshot();
	});

	it('should render correctly with disableTitleStyles prop', () => {
		const BreadCrumbs = () => <div>Breadcrumb</div>;
		const Actions = () => <div>Action</div>;
		const Bar = () => <div>Bar</div>;

		const wrapper = (
			<PageHeader
				breadcrumbs={<BreadCrumbs />}
				actions={<Actions />}
				bottomBar={<Bar />}
				disableTitleStyles
			>
				Test
			</PageHeader>
		);

		const Component = renderer.create(wrapper).toJSON();
		expect(Component).toMatchSnapshot();
	});

	it('should render component as <h1> level heading', () => {
		const { container } = render(<PageHeader>Title</PageHeader>);

		expect(container.querySelector('h1')).toBeInTheDocument();
	});

	it('should render passed children', () => {
		render(<PageHeader>Title</PageHeader>);

		expect(screen.getByText('Title')).toBeInTheDocument();
	});

	it('should render all passed children components', () => {
		const Bar = () => <div>Bottom bar</div>;
		const Actions = () => <div>Actions</div>;
		const BreadCrumbs = () => <div>Breadcrumb</div>;
		render(
			<PageHeader bottomBar={<Bar />} actions={<Actions />} breadcrumbs={<BreadCrumbs />}>
				Title
			</PageHeader>,
		);

		expect(screen.getByText('Bottom bar')).toBeInTheDocument();
		expect(screen.getByText('Actions')).toBeInTheDocument();
		expect(screen.getByText('Breadcrumb')).toBeInTheDocument();
		expect(screen.getByText('Title')).toBeInTheDocument();
	});

	it('should render custom component without the StyledTitle when disableTitleStyles is true', () => {
		const CustomTitle = () => <span>Custom component</span>;
		const { container } = render(
			<PageHeader disableTitleStyles>
				<CustomTitle />
			</PageHeader>,
		);

		expect(screen.getByText('Custom component')).toBeInTheDocument();
		expect(container.querySelector('h1')).not.toBeInTheDocument();
	});

	it('should truncate with truncateTitle prop', () => {
		render(<PageHeader truncateTitle>Long heading text</PageHeader>);

		const element = screen.getByText('Long heading text');

		expect(element).toHaveStyleDeclaration('white-space', 'nowrap');
		expect(element).toHaveStyleDeclaration('text-overflow', 'ellipsis');
		expect(element).toHaveStyleDeclaration('overflow-x', 'hidden');
	});

	it('should set received id prop as id of inner h1 element', () => {
		render(<PageHeader id="page-heading">Title</PageHeader>);

		expect(screen.getByText('Title')).toHaveAttribute('id', 'page-heading');
	});
});
