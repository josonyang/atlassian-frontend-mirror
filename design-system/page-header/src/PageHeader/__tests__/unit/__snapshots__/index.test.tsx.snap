// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@atlaskit/page-header should render correctly 1`] = `
.emotion-5 {
  margin: var(--ds-space-300, 24px) 0 var(--ds-space-200, 16px) 0;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-1 {
  min-width: 0;
  max-width: 100%;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  margin-block-end: var(--ds-space-100, 8px);
}

.emotion-0 {
  font-size: 1.7142857142857142em;
  font-style: inherit;
  line-height: 1.1666666666666667;
  color: var(--ds-text, #172B4D);
  font-weight: var(--ds-font-weight-medium, 500);
  letter-spacing: -0.01em;
  margin-top: 40px;
  line-height: var(--ds-font-lineHeight-500, 32px);
  margin-block-start: 0;
  outline: none;
}

.emotion-2 {
  max-width: 100%;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-block-end: var(--ds-space-100, 8px);
  -webkit-margin-start: auto;
  margin-inline-start: auto;
  -webkit-padding-start: var(--ds-space-400, 32px);
  padding-inline-start: var(--ds-space-400, 32px);
  white-space: nowrap;
}

.emotion-2 > {
  text-align: right;
}

.emotion-4 {
  box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border: none;
  margin-block-start: var(--ds-space-200, 16px);
}

<div
  className="emotion-5"
>
  <div>
    Breadcrumb
  </div>
  <div
    className="emotion-3"
  >
    <div
      className="emotion-1"
    >
      <h1
        className="emotion-0"
        tabIndex={-1}
      >
        Test
      </h1>
    </div>
    <div
      className="emotion-2"
    >
      <div>
        Action
      </div>
    </div>
  </div>
  <div
    className="emotion-4"
  >
     
    <div>
      Bar
    </div>
     
  </div>
</div>
`;

exports[`@atlaskit/page-header should render correctly with disableTitleStyles prop 1`] = `
.emotion-4 {
  margin: var(--ds-space-300, 24px) 0 var(--ds-space-200, 16px) 0;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-0 {
  min-width: 0;
  max-width: 100%;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  margin-block-end: var(--ds-space-100, 8px);
}

.emotion-1 {
  max-width: 100%;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-block-end: var(--ds-space-100, 8px);
  -webkit-margin-start: auto;
  margin-inline-start: auto;
  -webkit-padding-start: var(--ds-space-400, 32px);
  padding-inline-start: var(--ds-space-400, 32px);
  white-space: nowrap;
}

.emotion-1 > {
  text-align: right;
}

.emotion-3 {
  box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border: none;
  margin-block-start: var(--ds-space-200, 16px);
}

<div
  className="emotion-4"
>
  <div>
    Breadcrumb
  </div>
  <div
    className="emotion-2"
  >
    <div
      className="emotion-0"
    >
      Test
    </div>
    <div
      className="emotion-1"
    >
      <div>
        Action
      </div>
    </div>
  </div>
  <div
    className="emotion-3"
  >
     
    <div>
      Bar
    </div>
     
  </div>
</div>
`;
